✨ feat     : bookmark // recruitment api 개발

Zwonnyy opened the pull request:
✨ feat     : bookmark // recruitments api 개발

bookmark // recruitments api 개발

## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: bookmark // recruitments  기능개발 및 테스트코드 작성

## 📄 상세 내용
- [x] test_bookmark_api // test_recruitments_api 작성완료
- [x] bookmark // recruitments  기능개발

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Zwonnyy committed:
✨ feat     : bookmark // recruitments 기능 개발

bookmark // recruitments api 개발

choismjames23 commented on apps/recruitments/tests/test_bookmark_api.py:
```diff
@@ -0,0 +1,128 @@
+        )
+
+        # ✅ 실제 엔드포인트 기반으로 수정
+        self.list_url = "/api/v1/recruitments/bookmark/"
```
url을 하드코딩 하는 것이 아닌, reverse를 이용하여 url 파일에 작성해둔 내역을 가져오는 것이 좋아보입니다

Meoyoug commented on apps/recruitments/serializers/bookmark.py:
```diff
@@ -3,14 +3,36 @@
+        try:
+            recruitment = Recruitment.objects.get(id=recruitment_id)
+        except Recruitment.DoesNotExist:
+            raise serializers.ValidationError({"detail": "존재하지 않는 모집글입니다."})
```
객체 조회와 예외케이스 에러응답처리는 View나 Service에서 이루어져야할 것 같습니다

Meoyoug commented on apps/recruitments/serializers/bookmark.py:
```diff
@@ -3,14 +3,36 @@
+
+        # 중복 방지
+        if Bookmark.objects.filter(user=user, recruitment=recruitment).exists():
+            raise serializers.ValidationError({"detail": "이미 북마크한 모집글입니다."})
```
이것도 일종의 Service로직의 일부같은데 별도로 Service를 사용하지않는다면 View에서 먼저 검증 후에 create호출하게 해주세요

Zwonnyy committed:
💡 chore    : 피드백 수정 및 모델 추가
💡 chore    : serializer/bookmark.py 코드 수정 및 service/bookmark.py 작성, __init__.py 수정 및 전체 모델 수정

Zwonnyy committed:
💡 chore    : services 추가 커밋
💡 chore    : services 추가 커밋

서비스 파일이 안올라가서 다시 커밋합니다

2joonkim commented:
주원님 조퇴로 인해 내일 프론트 요청사항 반영되면 다시 체크해서 머지 진행하겠습니다~!!
